server/issue.go

```diff
@@ -28,43 +28,64 @@ func makePost(userId, channelId, message string) *model.Post {
 	}
 }
 
+func respondf(plugin *Plugin, status int, e error, mmuserID, jiraBotID, channelID, message string) (int, error) {
+	_ = plugin.API.SendEphemeralPost(mmuserID, makePost(jiraBotID, channelID, message))
+	if e != nil {
+		return status, e
+	}
+	return status, errors.New(message)
+}
+
 func httpAPITransitionIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
 	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
+	plugin := ji.GetPlugin()
+	jiraBotID := plugin.getUserID()
+	channelID := requestData.ChannelId
 
 	mattermostUserId := requestData.UserId
 	if mattermostUserId == "" {
-		return http.StatusUnauthorized, errors.New("not authorized")
+		return respondf(plugin, http.StatusUnauthorized, nil, mattermostUserId, jiraBotID, channelID, "user not authorized")
 	}
 
-	issueKey := requestData.Context["issueKey"].(string)
-	toState := requestData.Context["selected_option"].(string)
+	var issueKey string
+	var toState string
+	var found bool
+	var ok bool
+	var x interface{}
```

Comment:
We can remove these var declarations, based on my suggestions below.

Reply:
Worked on this.